src/textual/widgets/_option_list.py

```diff
@@ -0,0 +1,903 @@
+"""Provides the core of a classic vertical bounce-bar option list.
+
+Useful as a lightweight list view (not to be confused with ListView, which
+is much richer but uses widgets for the items) and as the base for various
+forms of bounce-bar menu.
+"""
+
+from __future__ import annotations
+
+from typing import ClassVar, Iterator, NamedTuple
+
+from rich.console import RenderableType
+from rich.repr import Result
+from rich.rule import Rule
+from rich.segment import Segment
+from rich.style import Style
+from typing_extensions import Literal, Self, TypeAlias
+
+from ..binding import Binding, BindingType
+from ..events import Click, MouseMove
+from ..geometry import Region, Size
+from ..message import Message
+from ..reactive import reactive
+from ..scroll_view import ScrollView
+from ..strip import Strip
+
+
+class DuplicateID(Exception):
+    """Exception raised if a duplicate ID is used."""
+
+
+class Option:
+    """Class that holds the details of an individual option."""
+
+    def __init__(
+        self, prompt: RenderableType, id: str | None = None, disabled: bool = False
+    ) -> None:
+        """Initialise the option.
+
+        Args:
+            prompt: The prompt for the option.
+            id: An optional ID for the option.
+            disabled: The initial enabled/disabled state. Enabled by default.
+        """
+        self.__prompt = prompt
```

Comment:
Folk will be encouraged to inherit from `Option` if they want to add extra data as cargo for options, but I want to discourage messing with the prompt under the hood in any child class as much as possible (because there is no `Option` -> `OptionList` coupling that would have the prompt update.